app/models/crate.js

```diff
@@ -41,6 +42,41 @@ export default class Crate extends Model {
     }
   }
 
+  @cached get versionIdsBySemver() {
+    let versions = this.versions.toArray() ?? [];
+    return versions.sort(compareVersionBySemver).map(v => v.id);
+  }
+
+  @cached get versionIdsByDate() {
+    let versions = this.versions.toArray() ?? [];
+    return versions.sort(compareVersionByDate).map(v => v.id);
+  }
+
+  @cached get firstVersionId() {
+    return this.versionIdsByDate.at(-1);
+  }
+
+  @cached get versionsObj() {
+    let versions = this.versions.toArray() ?? [];
+    return Object.fromEntries(versions.map(v => [v.id, v]));
+  }
```

Comment:
are these recomputing correctly when the content of `this.versions` changes?

Reply:
Yes, I assumed so. Would you like me to add a test for this functionality?

Reply:
depends on how much effort that requires. maybe try to write a test that visits the version page, goes to another page, pushes another version into the Ember Data store, visits the version page again and asserts that the new version is now also correctly displayed?

Reply:
Instead of revisiting the page, I used `findRecord` with `reload` param to ensure fresh data. I'm not sure if this is sufficient, though.

Reply:
yeah, that should also be sufficient 👍